packages/ra-core/src/core/Resource.tsx

```diff
@@ -10,10 +10,32 @@ export const Resource = (props: ResourceProps) => {
     return (
         <ResourceContextProvider value={name}>
             <Routes>
-                {Create && <Route path="create/*" element={<Create />} />}
-                {Show && <Route path=":id/show/*" element={<Show />} />}
-                {Edit && <Route path=":id/*" element={<Edit />} />}
-                {List && <Route path="/*" element={<List />} />}
+                {Create && (
+                    <Route
+                        path="create/*"
+                        element={
+                            React.isValidElement(Create) ? Create : <Create />
```

Comment:
Not related, but this allows to use elements in Resource:

```jsx
<Resource name="posts" list={<PostList />} />
```

In addition to using components:

```jsx
<Resource name="posts" list={PostList} />
```